### PR TITLE
fix/logs: Fix for nested log filters when set to false

### PR DESCRIPTION
**Pull request by Ziinc**
Fixes the instance when filter is toggled on and off, resulting in the following error. 

image

Error results from query building when nested queries are by default wrapped with a `()`, resulting in `... where () limit 100`. 

Additional test case has been added to query builder.

### CODE REVIEW

**Review by joshenlim — APPROVED** [review]
yeap lgtm!

**Comment by MildTomato** [comment]
@Ziinc made sure this works with filters being used *and* doing a text search?
Always a good manual test to do while making these changes.

that was usually the thing that went wrong with the SQL WHERE builder.

File: studio/components/interfaces/Settings/Logs/Logs.utils.ts [hunk 1]
```diff
@@ -1,135 +1,146 @@
...
  * The template of the filter determines the actual filter statement. If no template is provided, a generic equality statement will be used.
  * This only applies for root keys of the filter.
  * For example:
  * ```
  * {'my.nested.value': 123}
  * ```
  * with no template, it will be converted into `WHERE (my.nested.value = 123)
- * 
+ *
  * @returns a whewhere statement with WHERE clause.
```
Comment (kiwicopple) [inline studio/components/interfaces/Settings/Logs/Logs.utils.ts:43]:
```suggestion
 * @returns a where statement with WHERE clause.
```

**Review by kiwicopple — APPROVED** [review]

**Comment by Ziinc** [comment]
@MildTomato Yup, did a manual test to verify. Also just updated the test to consider concurrent search query. Also, compared to launch week implementation, the last refactor has combined the search filter generation together with normal filters, so it'll be less error prone now

**Pull request merged**